Source/objects.cpp

```diff
@@ -3654,163 +3624,167 @@ void OperateShrine(int pnum, int i, _sfx_id sType)
 		break;
 	case ShrineEldritch:
-		if (!OperateShrineEldritch(pnum))
+		if (!OperateShrineEldritch(player))
 			return;
 		break;
 	case ShrineEerie:
-		if (!OperateShrineEerie(pnum))
+		if (!OperateShrineEerie(player))
 			return;
 		break;
 	case ShrineDivine:
-		if (!OperateShrineDivine(pnum, Objects[i].position))
+		if (!OperateShrineDivine(player, shrine.position))
 			return;
 		break;
 	case ShrineHoly:
 		if (!OperateShrineHoly(pnum))
 			return;
 		break;
 	case ShrineSacred:
-		if (!OperateShrineSacred(pnum))
+		if (!OperateShrineSacred(player))
 			return;
 		break;
 	case ShrineSpiritual:
-		if (!OperateShrineSpiritual(pnum))
+		if (!OperateShrineSpiritual(player))
 			return;
 		break;
 	case ShrineSpooky:
-		if (!OperateShrineSpooky(pnum))
+		if (!OperateShrineSpooky(player))
 			return;
 		break;
 	case ShrineAbandoned:
-		if (!OperateShrineAbandoned(pnum))
+		if (!OperateShrineAbandoned(player))
 			return;
 		break;
 	case ShrineCreepy:
-		if (!OperateShrineCreepy(pnum))
+		if (!OperateShrineCreepy(player))
 			return;
 		break;
 	case ShrineQuiet:
-		if (!OperateShrineQuiet(pnum))
+		if (!OperateShrineQuiet(player))
 			return;
 		break;
 	case ShrineSecluded:
-		if (!OperateShrineSecluded(pnum))
+		if (!OperateShrineSecluded(player))
 			return;
 		break;
 	case ShrineOrnate:
-		if (!OperateShrineOrnate(pnum))
+		if (!OperateShrineOrnate(player))
 			return;
 		break;
 	case ShrineGlimmering:
-		if (!OperateShrineGlimmering(pnum))
+		if (!OperateShrineGlimmering(player))
 			return;
 		break;
 	case ShrineTainted:
-		if (!OperateShrineTainted(pnum))
+		if (!OperateShrineTainted(player))
 			return;
 		break;
 	case ShrineOily:
-		if (!OperateShrineOily(pnum, Objects[i].position))
+		if (!OperateShrineOily(player, shrine.position))
 			return;
 		break;
 	case ShrineGlowing:
-		if (!OperateShrineGlowing(pnum))
+		if (!OperateShrineGlowing(player))
 			return;
 		break;
 	case ShrineMendicant:
-		if (!OperateShrineMendicant(pnum))
+		if (!OperateShrineMendicant(player))
 			return;
 		break;
 	case ShrineSparkling:
-		if (!OperateShrineSparkling(pnum, Objects[i].position))
+		if (!OperateShrineSparkling(player, shrine.position))
 			return;
 		break;
 	case ShrineTown:
-		if (!OperateShrineTown(pnum, Objects[i].position))
+		if (!OperateShrineTown(pnum, shrine.position))
 			return;
 		break;
 	case ShrineShimmering:
-		if (!OperateShrineShimmering(pnum))
+		if (!OperateShrineShimmering(player))
 			return;
 		break;
 	case ShrineSolar:
-		if (!OperateShrineSolar(pnum))
+		if (!OperateShrineSolar(player))
 			return;
 		break;
 	case ShrineMurphys:
-		if (!OperateShrineMurphys(pnum))
+		if (!OperateShrineMurphys(player))
 			return;
 		break;
 	}
 
-	CalcPlrInv(Players[pnum], true);
+	CalcPlrInv(player, true);
 	force_redraw = 255;
 
 	if (pnum == MyPlayerId)
 		NetSendCmdParam2(false, CMD_PLROPOBJ, pnum, i);
 }
 
-void OperateSkelBook(int pnum, int i, bool sendmsg)
+void OperateSkelBook(Player &player, int i, bool sendmsg)

```

Comment:
These operate bookcase functions don't use the player instance beyond checking if a network message should be sent to say the object was interacted with. I haven't looked at the usage to see if the existing `sendmsg` parameter is appropriate, for these cases it's probably worth leaving the argument as `pnum` or making it a new bool instead of a player reference.

Reply:
Looks like this is a safe assumption, it can only be called from ParseCmd where `sendmsg` isn't clearly the same condition. And that should afaik always be coming from a different player id.